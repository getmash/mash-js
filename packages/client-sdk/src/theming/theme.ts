export type Theme = {
  primaryColor: string;
  fontFamily: string;
};
