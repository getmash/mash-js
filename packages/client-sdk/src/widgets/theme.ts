import { TinyColor } from "@ctrl/tinycolor";

import { Theme } from "../api/routes.js";

export default function injectTheme(baseUrl: string, theme: Theme) {
  const color = new TinyColor(theme.primaryColor);
  const hsl = color.toHsl();

  const style = window.document.createElement("style");
  style.innerHTML = `
:root {
  --mash-primary-color-h: ${hsl.h};
  --mash-primary-color-s: ${hsl.s * 100}%;
  --mash-primary-color-l: ${hsl.l * 100}%;
  --mash-primary-color: ${color.toHexString()};
  --mash-font-family: ${theme.fontFamily};
}  
`;
  window.document.head.appendChild(style);

  const link = window.document.createElement("link");
  link.rel = "stylesheet";
  link.href = baseUrl + "/theme/theme.css";

  window.document.head.appendChild(link);
}
