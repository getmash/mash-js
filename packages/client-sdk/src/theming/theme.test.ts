import { build } from "./theme";

describe("theme", () => {
  describe("build", () => {
    it("build theme with color: #000, should match snapshot", () => {
      const result = build("#000");
      expect(result).toMatchSnapshot();
    });
    it("build theme with color: #fff, should match snapshot", () => {
      const result = build("#fff");
      expect(result).toMatchSnapshot();
    });
  });
});
