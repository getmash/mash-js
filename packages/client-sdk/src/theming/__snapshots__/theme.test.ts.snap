// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme build build theme with color: #000, should match snapshot 1`] = `
"
:root {
  --mash-primary-color: #000000;
}

mash-button::part(button), mash-accordion::part(button) {
  font-family: inherit;
}
mash-button::part(solid), mash-accordion::part(solid) {
  background-color: var(--mash-primary-color);
  color: #FFF;
}
mash-button::part(solid):hover, mash-accordion::part(solid):hover {
  background-color: #000000;
}
mash-button::part(solid):active, mash-accordion::part(solid):active {
  background-color: #000000;
}
mash-button::part(outlined), mash-accordion::part(outlined) {
  border-color: var(--mash-primary-color);
  background-color: transparent;
  color: var(--mash-primary-color);
}
mash-button::part(outlined):hover, mash-accordion::part(outlined):hover {
  background-color: rgb(204, 204, 204);
}
mash-button::part(outlined):active, mash-accordion::part(outlined):active {
  background-color: rgb(153, 153, 153);
}"
`;

exports[`theme build build theme with color: #fff, should match snapshot 1`] = `
"
:root {
  --mash-primary-color: #ffffff;
}

mash-button::part(button), mash-accordion::part(button) {
  font-family: inherit;
}
mash-button::part(solid), mash-accordion::part(solid) {
  background-color: var(--mash-primary-color);
  color: #000;
}
mash-button::part(solid):hover, mash-accordion::part(solid):hover {
  background-color: #cccccc;
}
mash-button::part(solid):active, mash-accordion::part(solid):active {
  background-color: #999999;
}
mash-button::part(outlined), mash-accordion::part(outlined) {
  border-color: var(--mash-primary-color);
  background-color: transparent;
  color: var(--mash-primary-color);
}
mash-button::part(outlined):hover, mash-accordion::part(outlined):hover {
  background-color: rgb(255, 255, 255);
}
mash-button::part(outlined):active, mash-accordion::part(outlined):active {
  background-color: rgb(255, 255, 255);
}"
`;
